src/trio/_tests/test_exports.py

```diff
@@ -306,7 +306,8 @@ def lookup_symbol(symbol: str) -> dict[str, str]:
                     mod_cache = next_cache / "__init__.data.json"
                 else:
                     mod_cache = mod_cache / (modname[-1] + ".data.json")
-
```

Comment:
Bugfix for `os.Pathlike`

Reply:
would you mind elaborating on what the bug/fix is? This code should probably be commented cause reading it now I don't understand almost any of it x)

Reply:
Guess what... I don't really understand it either! 😢 

Here's what I do know... this code is mapping a module name to a file in the `mypy` cache dir. For modules, this file is `foo.data.json`. for packages, this file is `package/__init__.data.json`.

You'll see a few lines above this change that the case of a package is handled for things like `foo.bar.baz` (i.e. if `modname` is truthy). But the package case is not otherwise handled. Now it is.

I found this check extremely confusing in general. I can see what it's trying to do, but it was thrown utterly by my completely reasonable _subclassing of a stdlib class_. Having to figure out what this complex linter was doing and then put in `missing -= PurePath.__dict__.keys()` just to keep it happy about _inheritance_ was a good 45 mins of frustration! And then I had to do it all again for the case of windows-only methods with overrides specific to `jedi` and `mypy`! In this latter case, the only way to test it was a push-to-CI-and-wait cycle.

It's working now, but are we sure this linter is worth the maintenance burden? An opinionated `ruff` coupled with very strict `mypy` is already some pretty heavy-duty guardrails. This discussion is out of scope for the PR, I think though.

Reply:
They were introduced because we in fact have had errors several times, the logic for exported symbols is kinda overly complex. But for a PR like this that overhauls a complex part I do find it good to know that we have exhaustive tests that would catch if something got missed.
But the burden they impose on contributors is non-trivial, for sure. I can maybe try to add more comments to the code & improve error messages to make it easier to know how to handle new problems at least.

Reply:
It's definitely a call for core project maintainers to make! 🙇